app/models/vacancy.rb

```diff
@@ -216,4 +207,10 @@ def enable_job_applications_cannot_be_changed_once_listed
 
     errors.add(:enable_job_applications, :cannot_be_changed_once_listed)
   end
+
+  def drop_attributes_if_no_longer_applicable
```

Comment:
I'd like to draw reviewers attention to this method. When we remove a step, we should remove the attributes that are no longer permitted, in order for the data to be self-consistent (e.g. so we don't display something on the review step or the listing that is not applicable.)

Reply:
OK, this is now going to be changed to use the new Resettable module that @robyoung91 is creating in https://github.com/DFE-Digital/teaching-vacancies/pull/4070